core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -246,6 +251,38 @@ class BlockManagerMasterEndpoint(
     blockManagerIdByExecutor.get(execId).foreach(removeBlockManager)
   }
 
+  private def recoverLatestRDDBlock(
+      execId: String,
+      excludeExecutors: Seq[String],
+      context: RpcCallContext): Unit = {
+    logDebug(s"Replicating first cached block on $execId")
+    val excluded = excludeExecutors.flatMap(blockManagerIdByExecutor.get)
+    val response: Option[Future[Boolean]] = for {
+      blockManagerId <- blockManagerIdByExecutor.get(execId)
+      info <- blockManagerInfo.get(blockManagerId)
+      blocks = info.cachedBlocks.collect { case r: RDDBlockId => r }
```

Comment:
in one comment, you mention that you are only replicating in-memory rdd blocks.  But this will also replicate on-disk rdd blocks, I think.  do you want to replicate both?

Reply:
Updated this so we are only replicating in memory blocks.

Reply:
sorry my comment was worded very poorly.  Is there any reason you wouldn't want to transfer the on-disk blocks?  I assume you'd want to replicate all of them, and just needed to change the comments elsewhere.  Intentional tradeoff, as users are more likely to limit the amount of in-memory caching to only the most important stuff?

Also just to be really precise -- the check below isn't whether the block is in-memory currently, its whether its been requested to cache in memory.  It may have been cached as MEMORY_AND_DISK, but currently only resides on disk.  Depending on why you want to limit to in-memory only, this may not be applicable.  Maybe you actually want `!useDisk`

Reply:
Thanks for the clarification @squito . So my original vision for this feature was that we would just do in memory blocks, and that this would be more lightweight than all blocks, and still useful for spark notebook users who had persisted the results of some long calculation. However, maybe replicating all blocks would be more useful and then I could get rid of that .checkMem function and simplify the code a bit. 

Do you have any preference for which approach to take here? I'm open to input.

Thanks.

Reply:
I don't have a strong opinion, but it sounds like your original approach was motivated by a small set of just the in-memory cached blocks.  Lets stick to in-memory only blocks for this version (`!useDisk`) as that limits the blocks we try to replicated, and we can consider relaxing that in the future.

Reply:
ok cool thanks.